src/geanyobject.c

```diff
@@ -380,3 +389,13 @@ GObject *geany_object_new(void)
 {
 	return g_object_new(GEANY_OBJECT_TYPE, NULL);
 }
+
+/** Get the global GeanyObject instance
+ *
+ * @return @transfer{none} The instance
+ */
+GEANY_API_SYMBOL
+GObject *geany_object_get_instance(void)
```

Comment:
pygobject only provides the facility to use python functions for signal callbacks for g_signal_connect (it acually maps GObject.connect() internally to g_signal_connect_closure() with a custom closure that creates PyGObject instances from whatever is passed to the signal handlers. this is also one reason that signal callback parameters must have a associated GType).